api.py

```diff
@@ -412,7 +412,124 @@ def delete(self):
+                response['message'] = 'RA and DEC arrays must have same dimensions'
 
+        if response['status'] == 'ok':
+            if 'band' in arguments:
+                bands = arguments['band'].replace('[','').replace(']','').replace("'",'').replace(' ','')
+                bands_set = set(bands.lower().split(','))
+                default_set = set(['g', 'r', 'i', 'z', 'y'])
+                if bands_set.issubset(default_set):
+                    bands = bands.lower().replace('y', 'Y')
+                else:
+                    self.set_status(400)
+                    response['status']='error'
+                    response['message'] = "Optional band must be one of g, r, i, z, Y"                    
+            else:
+                bands = 'all' 
+
+        #options
+        if response['status'] == 'ok':
+
+            if 'no_blacklist' in arguments:
+                noBlacklist = arguments["no_blacklist"] == 'true'
+            else:
+                noBlacklist = False
+
+            if 'ccdnum' in arguments:
+                try:
+                    ccdnum = [int(i) for i in arguments['ccdnum'].replace('[','').replace(']','').split(',')]
+                    options['ccdnum'] = ccdnum
+                except:
+                    self.set_status(400)
+                    response['status']='error'
+                    response['message'] = "Invalid ccdnum!"  
+
+            if 'expnum' in arguments:
+                try:
+                    expnum = [int(i) for i in arguments['expnum'].replace('[','').replace(']','').split(',')]
+                    options['expnum'] = expnum
+                except:
+                    self.set_status(400)
+                    response['status']='error'
+                    response['message'] = "Invalid expnum!"  
+
+
+            if 'nite' in arguments:
+                try:
+                    nite = [int(i) for i in arguments['nite'].replace('[','').replace(']','').split(',')] 
+                    options['nite'] = nite
+                except:
+                    self.set_status(400)
+                    response['status']='error'
+                    response['message'] = "Invalid nite!" 
+        
+        if response['status'] == 'ok':
+
+            #init jobid
+            qTaskId = str(uuid.uuid4())
+            # put ra, dec into a datafram
+            df_pos = pd.DataFrame(np.array([ra,dec]).T,columns=['RA','DEC'])
+            now = datetime.datetime.now()
+            qTiid = user+'_mongo_'+qTaskId+'_{'+now.strftime('%a %b %d %H:%M:%S %Y')+'}'
+            print (options)
+            try:
+                dtasks.getList.apply_async(args=[df_pos, options, bands, noBlacklist], task_id=qTiid)
+            except Exception as e:
+                response['status']='error'
+                response['message']=str(e)
+            else:
+                res = AsyncResult(qTiid)
+                result = res.wait(5)
+                rt_str = io.StringIO(result)
```

Comment:
what are these 2 lines doing ? its writing the request into a file so celery can read it?